File: 1672-richest-customer-wealth/1672-richest-customer-wealth.java
class Solution {
  public int maximumWealth(int[][] accounts) {
      int richestWealth = 0;

      for (int[] customer : accounts) {
          int currentCustomerWealth = 0;
          for (int bank : customer) {
              currentCustomerWealth += bank;
          }
          
          richestWealth = Math.max(richestWealth, currentCustomerWealth);
          // if (temp_wealth > richest_wealth) {
          //     richestWealth = temp_wealth;
          // }
      }

      return richestWealth;
  }
}

// Runtime: 0 ms (100.00%), Memory: 42.36 MB (88.00%)
// Time Complexity: O(n*m), where n is the number of customers and m is the number of banks.
// Space Complexity: O(1), no extra space used.

File: 1672-richest-customer-wealth/1672-richest-customer-wealth.py
class Solution:
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        richest = 0
        for i in range(len(accounts)):
            wealth = 0
            for j in range(len(accounts[i])):
                wealth += accounts[i][j]
            
            if wealth > richest:
                richest = wealth

        return richest

# Runtime: 3 ms (5.68%), Memory: 19.49 MB (25.20%)
# Time complexity: O(n*m) where n is the number of customers and m is the number of accounts per customer
# Space complexity: O(1)
